app/lib/url.rb

```diff
@@ -7,7 +7,7 @@ def self.protocol
   end
 
   def self.domain
-    Rails.application&.initialized? ? SiteConfig.app_domain : ApplicationConfig["APP_DOMAIN"]
+    SiteConfig.respond_to?(:app_domain) ? SiteConfig.app_domain : ApplicationConfig["APP_DOMAIN"]
```

Comment:
This didn't work as we expected. Because we ended up calling this method from within `SiteConfig` itself, we hit a weird edge case where the original `Rails.application&.initialized?` logic was `true`, but `app_domain` hadn't been defined yet on `SiteConfig` so we ended up with an `undefined method` error when starting up the app.